python/pyspark/find_spark_home.py

```diff
@@ -42,7 +42,11 @@ def is_spark_home(path):
     import_error_raised = False
     from importlib.util import find_spec
     try:
+        # Spark distribution can be downloaded when HADOOP_VERSION environment variable is set.
+        # We should look up this directory first, see also SPARK-32017.
+        spark_dist_dir = "spark-distribution"
         module_home = os.path.dirname(find_spec("pyspark").origin)
+        paths.append(os.path.join(module_home, spark_dist_dir))
```

Comment:
We should insert this entry between `"../"` and `os.path.dirname(os.path.realpath(__file__))`, which is the module home; otherwise, `os.path.dirname(os.path.realpath(__file__))` will be the spark home.

Reply:
`os.path.dirname(os.path.realpath(__file__))` is usually in a script directory because `find_spark_home.py` is included as a script at `setup.py`: https://github.com/apache/spark/blob/058e61ae143a3619fc13fb0c316044a75f7bc15f/python/setup.py#L187

for example, it will be user's `bin`.

I thought `os.path.dirname(os.path.realpath(__file__))` is more just dev purpose. I checked that it correctly points out the newly installed Spark home.

However, sure. why not be safer :-) I will update.

Reply:
The function in the file `find_spark_home.py` is called from `launch_gateway`, so if the users initialize Spark by themselves in their Python repr, the path will be the module home?

```py
$ python
>>> from pyspark.sql import SparkSession
>>> spark = SparkSession.builder.getOrCreate()
```